snapshot/restore/local_fs_output.go

```diff
@@ -403,6 +404,14 @@ func (o *FilesystemOutput) copyFileContent(ctx context.Context, targetPath strin
 	}
 	defer r.Close() //nolint:errcheck
 
+	type withProgressTracking interface {
```

Comment:
I think it's easier to wrap "r" so that it reports bytes written in the callback.

Reply:
Thank you for the idea, I've changed the code, is it what you meant ?